crates/chain/src/tx_graph.rs

```diff
@@ -449,6 +449,32 @@ impl<A> TxGraph<A> {
     pub fn is_empty(&self) -> bool {
         self.txs.is_empty()
     }
+
+    /// Map TxGraph for another anchor type
+    pub fn map_anchors<A2: Ord, F>(self, f: F) -> TxGraph<A2>
+    where
+        F: Fn(A) -> A2,
```

Comment:
I think using `FnMut` will be better. This gives the caller more flexibility.

Also note that [`Iterator::map` takes in `FnMut`](https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.map).

Reply:
We can not use `FnMut` here, and we use `FnOnce` which also gives caller the flexibility.